### PR TITLE
Add support for join raid protection

### PR DESCRIPTION
**Pull request by lukellmann**
see https://github.com/discord/discord-api-docs/pull/5778 and https://github.com/discord/discord-api-docs/pull/6133

### CODE REVIEW

File: core/src/commonMain/kotlin/cache/data/GuildData.kt [hunk 1]
```diff
@@ -65,7 +65,8 @@ public data class GuildData(
     val stageInstances: Optional<List<StageInstanceData>> = Optional.Missing(),
     val stickers: Optional<List<StickerData>> = Optional.Missing(),
     val guildScheduledEvents: Optional<List<GuildScheduledEventData>> = Optional.Missing(),
-    val premiumProgressBarEnabled: Boolean
+    val premiumProgressBarEnabled: Boolean,
+    val safetyAlertsChannelId: Snowflake?,
```
Comment (HopeBaron) [inline core/src/commonMain/kotlin/cache/data/GuildData.kt:69]: It's weird this is required to be filled, according to how the docs describe it this should be optional

File: rest/src/commonMain/kotlin/json/request/GuildRequests.kt [hunk 2]
```diff
@@ -244,6 +244,8 @@ public data class GuildModifyRequest(
     @SerialName("preferred_locale")
     val preferredLocale: Optional<String?> = Optional.Missing(),
     val features: Optional<Set<GuildFeature>> = Optional.Missing(),
+    @SerialName("safety_alerts_channel_id")
+    val safetyAlertsChannelId: OptionalSnowflake? = OptionalSnowflake.Missing,
```
Comment (HopeBaron) [inline rest/src/commonMain/kotlin/json/request/GuildRequests.kt:248]: You set it as optional here, but as a required nullable in the guild data object?

Please refer to my comment above

File: core/src/commonMain/kotlin/cache/data/GuildData.kt (hunk 1 above)
Comment (lukellmann) [reply core/src/commonMain/kotlin/cache/data/GuildData.kt:69]: in the guild object it's described to only be nullable: https://discord.com/developers/docs/resources/guild#guild-object-guild-structure (`?` before the type means nullable)

File: rest/src/commonMain/kotlin/json/request/GuildRequests.kt (hunk 2 above)
Comment (lukellmann) [reply rest/src/commonMain/kotlin/json/request/GuildRequests.kt:248]: yeah, this is the modify request:
missing means leave it as is, null means reset it

https://discord.com/developers/docs/resources/guild#modify-guild-json-params says it's optional and nullable (all parameters are optional as described in the info box above)

**Comment by lukellmann** [comment]
could you take another look @HopeBaron?

**Review by DRSchlaubi — APPROVED** [review]

**Pull request merged**